build_tools/github_action/configure_ci.py

```diff
@@ -240,11 +241,36 @@ def matrix_generator(
 
     if is_push and base_args.get("branch_name") == "main":
         print(f"[PUSH - MAIN] Generating build matrix with {str(base_args)}")
-        # Add all options
+        # Add all options except for families that allow failures
         for key in amdgpu_family_info_matrix:
-            if "linux" in amdgpu_family_info_matrix[key] and platform == "linux":
+            if (
+                "linux" in amdgpu_family_info_matrix[key]
+                and platform == "linux"
+                and "expect_failure" not in amdgpu_family_info_matrix[key]["linux"]
+            ):
                 potential_targets.append(key)
-            if "windows" in amdgpu_family_info_matrix[key] and platform == "windows":
+            if (
+                "windows" in amdgpu_family_info_matrix[key]
+                and platform == "windows"
+                and "expect_failure" not in amdgpu_family_info_matrix[key]["windows"]
+            ):
+                potential_targets.append(key)
+
+    if is_schedule:
+        print(f"[SCHEDULE] Generating build matrix with {str(base_args)}")
+        # Add all options that allow failures
+        for key in amdgpu_family_info_matrix:
+            if (
+                "linux" in amdgpu_family_info_matrix[key]
+                and platform == "linux"
+                and "expect_failure" in amdgpu_family_info_matrix[key]["linux"]
+            ):
+                potential_targets.append(key)
+            if (
+                "windows" in amdgpu_family_info_matrix[key]
+                and platform == "windows"
+                and "expect_failure" in amdgpu_family_info_matrix[key]["windows"]
+            ):
                 potential_targets.append(key)
```

Comment:
This logic is getting too complicated IMO. How about forking https://github.com/ROCm/TheRock/blob/main/build_tools/github_action/amdgpu_family_matrix.py into two files, one for per-commit runs (`push`, `pull_request`) runs and one for scheduled runs (`schedule`)? The `workflow_dispatch` trigger could choose either.

That would make it clear from the files what configurations run where, without needing to know "if expect_failure is set then it runs on these triggers but not these other triggers".

There are a few ways we could structure that:

* Full copy/paste, with duplication
* Common definition that both files pull from
* One as a subset of the other